src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/ReaderBigEndian.cs

```diff
@@ -9,6 +9,20 @@ namespace System.Buffers.Binary
 {
     public static partial class BinaryPrimitives
     {
+        /// <summary>
+        /// Reads a Double out of a read-only span of bytes as big endian.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double ReadDoubleBigEndian(ReadOnlySpan<byte> source)
+        {
+            double result = MemoryMarshal.Read<double>(source);
```

Comment:
What does the codegen on this look like? I would think a more optimized version would be:

```cs
return (BitConverter.IsLittleEndian)
    ? BitConverter.Int64BitsToDouble(MemoryMarshal.Read<long>(source))
    : MemoryMarshal.Read<double>(source);
```

(Same comment for other implementations.)

Reply:
According to sharplab:
```
C.ReadDoubleBigEndianOrig(System.ReadOnlySpan`1<Byte>)
    L0000: sub rsp, 0x38
    L0004: vzeroupper
    L0007: mov rax, [rcx]
    L000a: mov ecx, [rcx+0x8]
    L000d: cmp ecx, 0x8
    L0010: jl L0034
    L0012: vmovsd xmm0, qword [rax]
    L0016: vmovsd [rsp+0x30], xmm0
    L001c: mov rcx, [rsp+0x30]
    L0021: bswap rcx
    L0024: mov [rsp+0x28], rcx
    L0029: vmovsd xmm0, qword [rsp+0x28]
    L002f: add rsp, 0x38
    L0033: ret
    L0034: mov ecx, 0x28
    L0039: call System.ThrowHelper.ThrowArgumentOutOfRangeException(System.ExceptionArgument)
    L003e: int3

C.ReadDoubleBigEndianLevi(System.ReadOnlySpan`1<Byte>)
    L0000: sub rsp, 0x28
    L0004: vzeroupper
    L0007: mov rax, [rcx]
    L000a: mov ecx, [rcx+0x8]
    L000d: cmp ecx, 0x8
    L0010: jl L0028
    L0012: mov rcx, [rax]
    L0015: bswap rcx
    L0018: mov [rsp+0x20], rcx
    L001d: vmovsd xmm0, qword [rsp+0x20]
    L0023: add rsp, 0x28
    L0027: ret
    L0028: mov ecx, 0x28
    L002d: call System.ThrowHelper.ThrowArgumentOutOfRangeException(System.ExceptionArgument)
    L0032: int3
```

Reply:
At first I was just following the existing code, but I have updated the PR to match the suggestion.